ChatQnA/kubernetes/intel/README_tdx.md

```diff
@@ -0,0 +1,167 @@
+# Deploy ChatQnA in Kubernetes Cluster on Xeon with Intel TDX
+
+This document outlines the deployment process for a ChatQnA application utilizing the [GenAIComps](https://github.com/opea-project/GenAIComps.git) microservice pipeline components on Intel Xeon server where the microservices are protected by [Intel TDX](https://www.intel.com/content/www/us/en/developer/tools/trust-domain-extensions/overview.html).
+The guide references the project [GenAIInfra](https://github.com/opea-project/GenAIInfra.git) to prepare the infrastructure. 
+
+The deployment process is intended for users who want to deploy ChatQnA services:
+
+- with pods protected by Intel TDX,
+- on a single node in a cluster (acting as a master and worker) that is a Xeon 5th Gen platform or later,
+- running Ubuntu 24.04,
+- using images pushed to public repository, like quay.io or docker hub.
+
+It's split into 3 sections:
+
+1. [Cluster Configuration](#cluster-configuration) - steps required to prepare components in the cluster required to use Intel TDX.
+2. [Node configuration](#node-configuration) - additional steps to be performed on the node that are required to run heavy applications like OPEA ChatQnA.
+3. [ChatQnA Services Configuration and Deployment](#chatqna-services-configuration-and-deployment) - describes how to deploy ChatQnA services with Intel TDX protection.
+
+> [!NOTE]
+> Running TDX-protected services requires the user to define the pod's resources request (cpu, memory).
+>
+> Due to lack of hotplugging feature in TDX, the assigned resources cannot be changed after the pod is scheduled and the resources will not be shared with any other pod.
+>
+> This means, that the total amount of resources assigned to all TDX-protected pods must be less than the total amount of resources available on the node, leaving room for the non-TDX pods requests.
+
+
+## Cluster Configuration
+
+To prepare cluster to run Intel TDX-protected workloads, follow [Intel Confidential Computing Documentation](https://cc-enabling.trustedservices.intel.com/intel-confidential-containers-guide/01/introduction/index.html).
+
+
+## Node Configuration
+
+
+### Kubelet Configuration
+
+To run a complex and heavy application like OPEA, the cluster administrator must increase the kubelet timeout for container creation, otherwise the pod creation may fail due to timeout `Context deadline exceeded`.
+This is required because the container creation process can take a long time due to the size of pod images and the need to download the AI models.
```

Comment:
Is this timeout change generally required for any k8s deployment? If so should this be added to the main k8s readme?

Reply:
Is that just for peer pods or running CoCo on the host also often breaks 2 minutes?